### PR TITLE
add chainprenet tool from ucsc/kent

### PR DESCRIPTION
**Pull request by martenson**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/ucsc_tools/chainprenet/chainprenet.xml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                <param name="que_ref_index_file" type="select" label="Using reference genome index">
+                  <options from_data_table="fasta_indexes" />
+                </param>
+            </when>
+            <when value="history">
+                <param name="in_que_ref_index" format="tabular" type="data" label="Dataset with chrom sizes for query sequence. Typically a '.fai' file." />
+            </when>
+        </conditional>
+
+        <param argument="-dots" type="integer" optional="true" label="Output a dot every so often" />
```
Comment (bgruening) [inline tools/ucsc_tools/chainprenet/chainprenet.xml:70]: min=0?

File: tools/ucsc_tools/chainprenet/chainprenet.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        <param argument="-pad" type="integer" optional="true" label="Extra to pad around blocks to decrease trash" help="Default 1" />
```
Comment (bgruening) [inline tools/ucsc_tools/chainprenet/chainprenet.xml:71]: min=0

**Review by bgruening — COMMENTED** [review]
A test using the datatables would be ace.

File: tools/ucsc_tools/chainprenet/chainprenet.xml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+<tool id="chainprenet" name="chainPreNet" version="@TOOL_VERSION@+galaxy0" profile="21.05" license="MIT">
```
[the remark this answers is not in the document]
Comment (martenson) [reply tools/ucsc_tools/chainprenet/chainprenet.xml:1]: sure thing!

File: tools/ucsc_tools/chainprenet/chainprenet.xml (hunk 1 above)
Comment (martenson) [reply tools/ucsc_tools/chainprenet/chainprenet.xml]: I dropped this, it only writes "progress" dots to stderr

File: tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    </inputs>
+    <outputs>
+        <data name="out" format="chain" />
+    </outputs>
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="in_chain" value="input.chain" />
+            <param name="pad" value="1000" />
+            <param name="target_reference_index_source_selector" value="history" />
+            <param name="in_tar_ref_index" value="hg38.chr20.chunk.fasta.fai"/>
```
Comment (bgruening) [inline tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml:79]:
```suggestion
            <conditional name="target_reference_index_source"/>
                <param name="target_reference_index_source_selector" value="history" />
                <param name="in_tar_ref_index" value="hg38.chr20.chunk.fasta.fai"/>
            </conditional>
```

File: tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            <param name="query_reference_index_source_selector" value="history" />
+            <param name="in_que_ref_index" value="mm39.chr2.chunk.fasta.fai"/>
```
Comment (bgruening) [inline tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml:81]:
```suggestion
            <conditional name="query_reference_index_source">
                <param name="query_reference_index_source_selector" value="history" />
                <param name="in_que_ref_index" value="mm39.chr2.chunk.fasta.fai"/>
            </conditional>
```

File: tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+            <output name="out" file="out.chain"/>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="in_chain" value="input.chain" />
+            <param name="pad" value="1000" />
+            <param name="target_reference_index_source_selector" value="cached" />
+            <param name="tar_ref_index_file" value="hg38.chr20.chunk"/>
+            <param name="query_reference_index_source_selector" value="cached" />
+            <param name="que_ref_index_file" value="mm39.chr2.chunk"/>
```
Comment (bgruening) [inline tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml:90]:
```suggestion
            <conditional name="target_reference_index_source"/>
                <param name="target_reference_index_source_selector" value="cached" />
                <param name="tar_ref_index_file" value="hg38.chr20.chunk"/>
            </conditional>
            <conditional name="query_reference_index_source">
                <param name="query_reference_index_source_selector" value="cached" />
                <param name="que_ref_index_file" value="mm39.chr2.chunk"/>
            </conditional>
```

**Review by bgruening — APPROVED** [review]

File: tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+    </inputs>
+    <outputs>
+        <data name="out" format="chain" />
+    </outputs>
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="in_chain" value="input.chain" />
+            <param name="pad" value="1000" />
+            <conditional name="target_reference_index_source"/>
```
Comment (SaimMomin12) [inline tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml:78]:
```suggestion
            <conditional name="target_reference_index_source">
```

File: tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            <conditional name="query_reference_index_source">
+                <param name="query_reference_index_source_selector" value="history" />
+                <param name="in_que_ref_index" value="mm39.chr2.chunk.fasta.fai"/>
+            </conditional>
+            <output name="out" file="out.chain"/>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="in_chain" value="input.chain" />
+            <param name="pad" value="1000" />
+            <conditional name="target_reference_index_source"/>
```
Comment (SaimMomin12) [inline tools/ucsc_tools/ucsc_chainprenet/chainprenet.xml:91]:
```suggestion
            <conditional name="target_reference_index_source">
```

**Review by SaimMomin12 — CHANGES_REQUESTED** [review]
@martenson There are two minor linting errors in the `chainprenet.xml` file. The changes can be accepted to pass the CI tests.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
That was my fault.

**Pull request merged**